siliconcompiler/core.py

```diff
@@ -4136,21 +4136,43 @@ def run(self):
             running_tasks = []
             while len(tasks_to_run) > 0 or len(running_tasks) > 0:
                 # Check for new tasks that can be launched.
-                for task, deps in list(tasks_to_run.items()):
+                for node, deps in list(tasks_to_run.items()):
                     # TODO: breakpoint logic:
                     # if task is breakpoint, then don't launch while len(running_tasks) > 0
 
+                    dep_was_successful = False
+                    had_deps = len(deps) > 0
+                    tool, task = self._get_tool_task(
+                        processes[node]._args[0],
+                        processes[node]._args[1])
```

Comment:
it would be nice to avoid reaching into the processing object for this information and instead track it ourselves.

Reply:
Noah had similar feedback see my comment here:
https://github.com/siliconcompiler/siliconcompiler/pull/1806#discussion_r1294336467
tldr: either I add another dict or we index the status dict by step, index

Reply:
separate PR is fine